File: appengine/flexible/Stackdriver/Controllers/HomeController.cs
/*
 * Copyright (c) 2017 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */

using Microsoft.AspNetCore.Mvc;
using Stackdriver.ViewModels;
using System.Threading.Tasks;
using Microsoft.Extensions.Options;

namespace Stackdriver.Controllers
{
    public class HomeController : Controller
    {
        readonly string _projectId;
        public HomeController(IOptions<StackdriverOptions> options)
        {
            _projectId = options.Value.ProjectId;
        }

        [HttpGet]
        public IActionResult Index()
        {
            var model = new HomeIndex() { ProjectId = _projectId };
            return View(model);
        }

        public IActionResult Error()
        {
            return View();
        }
    }
}

File: appengine/flexible/F#/Controllers/HomeController.fs
//
// Copyright (c) 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License"); you may not
// use this file except in compliance with the License. You may obtain a copy of
// the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
// License for the specific language governing permissions and limitations under
// the License.
//
namespace FSharp.Controllers

open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc
open Microsoft.AspNetCore.Http
open Google.Cloud.Vision.V1

type HomeController () =
    inherit Controller()

    member this.Index () =
        this.View()

    member this.About () =
        this.ViewData.["Message"] <- "Your application description page."
        this.View()

    member this.Contact () =
        this.ViewData.["Message"] <- "Your contact page."
        this.View()

    member this.Error () =
        this.View()

    member this.Naughty () =
        this.View()

    [<HttpPost>]
    member this.Naughty (files:List<IFormFile>) =
        match Seq.toList files with
        | [file] ->
            let vision = ImageAnnotatorClient.Create()
            let response = vision.DetectSafeSearch(Image.FromStream(file.OpenReadStream()))
            this.View((file, response))
        | _ -> this.View()